fe/fe-core/src/main/java/com/starrocks/qe/ShortCircuitExecutor.java

```diff
@@ -86,13 +88,16 @@ public void exec() {
         throw new StarRocksPlannerException("Not implement ShortCircuit Executor class", ErrorType.INTERNAL_ERROR);
     }
 
-    protected static <T> T pick(List<T> collections) {
-        Preconditions.checkArgument(!collections.isEmpty());
-        if (collections.size() == 1) {
-            return collections.get(0);
-        } else {
-            return collections.get(RANDOM.nextInt(collections.size()));
+    protected static Optional<TNetworkAddress> pick(List<com.starrocks.thrift.TNetworkAddress> hosts,
+                                                    List<Backend> aliveBackends) {
+        for (Backend be : aliveBackends) {
```

Comment:
it will be less efficient if the cluster size is large with hundreds of nodes. may consider build a map for the alive backends and shuffle the `hosts` for the scan range.

Reply:
this pr mainly resolve query request can't send to be when the be is not alive . if using the map, will need listen the status of nodes. the shuffle of hundreds of nodes maybe not cost too much?

Reply:
I mean just rebuild the backend as map instead of list. so it can be quickly filtered instead of linear search.